src/modules/ekf2/EKF/ekf.cpp

```diff
@@ -335,9 +335,11 @@ bool Ekf::resetGlobalPosToExternalObservation(const double latitude, const doubl
 
 		const bool innov_rejected = (test_ratio > 1.f);
 
-		if (!_control_status.flags.in_air || (eph > 0.f && eph < 1.f) || innov_rejected) {
-			// when on ground or accuracy chosen to be very low, we hard reset position
+		if (!_control_status.flags.in_air || (eph > 0.f && eph < 1.f) || innov_rejected || isHorizontalPositionAidingActive()) {
```

Comment:
this isn't required in this PR as we found another solution here: https://github.com/PX4/PX4-Autopilot/pull/25223/files#diff-758f8b424acb392d122caf1185c53f3f0b9304351ab11a85f1aecad89776a4b6R350-R371